overlord/ifacestate/helpers.go

```diff
@@ -348,6 +348,28 @@ func (m *InterfaceManager) reloadConnections(snapName string) ([]string, error)
 		return nil, err
 	}
 
+	var policyChecker interfaces.PolicyFunc
+	var autoChecker *autoConnectChecker
+	var connChecker *connectChecker
+
+	deviceCtx, err := snapstate.DeviceCtx(m.state, nil, nil)
```

Comment:
I've investigated why upgrade-from-2.15 fails with the branch. Turns out, there is no device context on the first run (i.e. when we start the new snapd for the first time), hence no connections re reloaded and the profiles are regenerated without any of the plugs connected.